OpenSim/Simulation/Model/GeometryPath.cpp

```diff
@@ -1278,10 +1269,10 @@ calcLengthAfterPathComputation(const SimTK::State& s,
 double GeometryPath::
 computeMomentArm(const SimTK::State& s, const Coordinate& aCoord) const
 {
-    if(!_maSolver)
-        _maSolver = new MomentArmSolver(*_model);
+    if (!_maSolver)
+        const_cast<Self*>(this)->_maSolver.reset(new MomentArmSolver(*_model));
```

Comment:
That `const_cast` seems unfortunate! Isn't there a way to make sure that the MomentArmSolver gets created earlier, in one of the methods that has write access to the GeometryPath? It looks like it just needs to get done after `_model` first gets set.